internal/pkg/checkin/bulk.go

```diff
@@ -208,6 +208,11 @@ func (bc *Bulk) flush(ctx context.Context) error {
 			if pendingData.extra.ver != "" {
 				fields[dl.FieldAgent] = map[string]interface{}{
 					dl.FieldAgentVersion: pendingData.extra.ver,
+					// Duplicates the writes done when an upgrade is ack'd by the agent.
+					// This helps eliminate the case where the agent is stuck in upgrading after it's already upgraded if the ack
+					// is never received.
+					dl.FieldUpgradeStartedAt: nil,
+					dl.FieldUpgradedAt:       nowTimestamp,
```

Comment:
it's not really clear how to assert in the unit test that extra fields were set, can we capture the args that the mock was called with in go? https://github.com/elastic/fleet-server/blob/c8966edac2e560671c9b2caabc9beda25bafacaa/internal/pkg/checkin/bulk_test.go#L184